src/main/scala/org/ergoplatform/network/ErgoSyncTracker.scala

```diff
@@ -22,34 +21,37 @@ final case class ErgoSyncTracker(system: ActorSystem,
                                  timeProvider: TimeProvider)
  extends ScorexLogging {
 
-  val MinSyncInterval: FiniteDuration = 20.seconds
-  val SyncThreshold: FiniteDuration = 1.minute
+  private val MinSyncInterval: FiniteDuration = 20.seconds
+  private val SyncThreshold: FiniteDuration = 1.minute
 
-  private[network] val statuses = mutable.Map[ConnectedPeer, ErgoPeerStatus]()
-  private[network] val lastSyncSentTime = mutable.Map[ConnectedPeer, Time]()
-  private[network] val lastSyncGetTime = mutable.Map[ConnectedPeer, Time]()
+  protected[network] val statuses: mutable.Map[ConnectedPeer, ErgoPeerStatus] =
+    mutable.Map[ConnectedPeer, ErgoPeerStatus]()
 
-  protected var lastSyncInfoSentTime: Time = 0L
-
-  val heights: mutable.Map[ConnectedPeer, Height] = mutable.Map[ConnectedPeer, Height]()
+  def fullInfo(): Iterable[ErgoPeerStatus] = statuses.values
 
   // returns diff
   def updateLastSyncGetTime(peer: ConnectedPeer): Long = {
-    val prevSyncGetTime = lastSyncGetTime.getOrElse(peer, 0L)
+    val prevSyncGetTime = statuses.get(peer).flatMap(_.lastSyncGetTime).getOrElse(0L)
     val currentTime = timeProvider.time()
-    lastSyncGetTime(peer) = currentTime
+    statuses.get(peer).foreach { status =>
+      statuses.update(peer, status.copy(lastSyncGetTime = Option(currentTime)))
+    }
     currentTime - prevSyncGetTime
   }
 
-  def fullInfo: Iterable[ErgoPeerStatus] = statuses.values
-
-  def isOutdated(peer: ConnectedPeer): Boolean = {
-    (timeProvider.time() - lastSyncSentTime.getOrElse(peer, 0L)).millis > SyncThreshold
+  def notSyncedOrOutdated(peer: ConnectedPeer): Boolean = {
+    val peerOpt = statuses.get(peer)
+    val notSyncedOrMissing = peerOpt.forall(_.lastSyncSentTime.isEmpty)
+    val outdated =
+      peerOpt
+        .flatMap(_.lastSyncSentTime)
+        .exists(syncTime => (timeProvider.time() - syncTime).millis > SyncThreshold)
+    notSyncedOrMissing || outdated
   }
 
   def updateStatus(peer: ConnectedPeer, status: HistoryComparisonResult, height: Option[Height]): Unit = {
     val seniorsBefore = numOfSeniors()
-    statuses += peer -> ErgoPeerStatus(peer, status, height.getOrElse(ErgoHistory.EmptyHistoryHeight))
+    statuses += peer -> ErgoPeerStatus(peer, status, height.getOrElse(ErgoHistory.EmptyHistoryHeight), None, None)
```

Comment:
Why None, None here for sync update time ? Sync times being erased on e.g. height update then

Reply:
Yes, you are right. fixed